src/service/getCreditScore.js

```diff
@@ -4,70 +4,155 @@ const {getAccountHistoryHoldings, calculateAccountHistoryHoldingsScore} = requir
 const {getPoaps, calculatePoapsCreditScore} = require("./getPoaps");
 const {retryIfFailed} = require("./retryService");
 const {getAaveAddressDetails, calculateAaveAddressDetailsScore} = require("./getAaveAddressDetails");
+const {GetIdentityList, average, calculateTwitterScore, GetFollowTwitterList} = require("./FollowersFollowingsScore");
+
+
+function getMeanOfFriendsSocialScore(address) {
+    /**
+     Returns friends social score mean of particular address
+     */
+
+    const identity_list = GetIdentityList(address, " followingCount\n followerCount\n friends{list{address\n}}\n");
+
+    return identity_list.then(v => {
```

Comment:
logicznie - o to chodziło. tyle że trochę ciężko ogarnąć ten kod - chodzi o mutowanie tablic/dodatkowe zmienne. Mozna to zrobić prościej funkcyjnie. spoko zasadą zeby zobaczyć czy Twój kod da się uprościć jest zobaczyć czy używasz const czy let/var (99% przypadków da się const i jest czyściej), to samo z pushem - 99% przypadków można go wywalić

```
function getMeanOfFriendsSocialScore(address) {
    /**
     Returns friends social score mean of particular address
     */
    const identityListPromise = GetIdentityList(address, " followingCount\n followerCount\n friends{list{address\n}}\n");

    return identityListPromise.then(identityList => {
        const friendsCreditStorePromiseList = identityList.follows.friends.list
            .map(friend => getCreditScore(friend.address, true));
        return Promise.all(friendsCreditStorePromiseList)
            .then(friendCreditScoreList => average(friendCreditScoreList.map(creditScore => creditScore.basicScore)));
    })
}
```
i w tym drugim pliku zmiana funkcji average:

`const average = (array) => array.reduce((a, b) => a + b, 0) / array.length;`